pipeline_steps/common.groovy

```diff
@@ -909,8 +909,29 @@ void standard_job_slave(String slave_type, Closure body){
         body()
       }
     } else if (slave_type == "container"){
-      dir("rpc-gating"){
-        container = docker.build env.BUILD_TAG.toLowerCase()
+      String image_name = env.BUILD_TAG.toLowerCase()
+      String dockerfile_repo_dir = "${env.WORKSPACE}/"
+
+      if (env.SLAVE_CONTAINER_DOCKERFILE_REPO == "PROJECT") {
+        if ( env.ghprbPullId != null ) {
+          clone_with_pr_refs(
+            "${env.WORKSPACE}/${env.RE_JOB_REPO_NAME}",
+          )
+        } else {
+          clone_with_pr_refs(
+            "${env.WORKSPACE}/${env.RE_JOB_REPO_NAME}",
+            env.REPO_URL,
+            env.BRANCH,
+          )
+        }
+
+        dockerfile_repo_dir += env.RE_JOB_REPO_NAME
+      } else {
```

Comment:
I think I'd run into an issue gating this change because the jobs will use this change itself but `env.SLAVE_CONTAINER_DOCKERFILE_REPO` won't have been set yet in their job configs.  I can either add a follow-up to do this once this change merges, or temporarily set `env.SLAVE_CONTAINER_DOCKERFILE_REPO` if it doesn't exist.